### ISSUE TITLE
incorrect IR generation

### ISSUE THREAD

**Issue by shubham162019** [openvinotoolkit/openvino#11483]
system info:- Ubuntu 20.04, python 3.8

I have used the model from :- https://github.com/xuannianz/DifferentiableBinarization.git
Using the code below, I load the model weights:- 

    `import sys
    sys.path.append("DifferentiableBinarization/")
    from DifferentiableBinarization.model import dbnet
    import keras.backend as K
    import tensorflow as tf
    from tensorflow import keras
    #from openvino.inference_engine import IENetwork, IECore
    #import cv2
    #import numpy as np
    #import time
    #import math
    #from shapely.geometry import Polygon
    #import pyclipper
    
    def freeze_session(session, keep_var_names=None, output_names=None, clear_devices=True):
        graph = session.graph
        with graph.as_default():
            freeze_var_names = list(set(v.op.name for v in tf.global_variables()).difference(keep_var_names or []))
            output_names = output_names or []
            output_names += [v.op.name for v in tf.global_variables()]
            input_graph_def = graph.as_graph_def()
            if clear_devices:
                for node in input_graph_def.node:
                    node.device = ''
            frozen_graph = tf.graph_util.convert_variables_to_constants(
                session, input_graph_def, output_names, freeze_var_names)
            return frozen_graph
    
    _, model = dbnet()
    K.set_learning_phase(0)
    model.load_weights('../Downloads/db_48_2.0216_2.5701.h5', by_name=True, skip_mismatch=True)
    #create an onnx model
    output_path = model.name + ".onnx"
    import tf2onnx
    import onnxruntime as rt
    spec = (tf.TensorSpec((None, 736, 640, 3), tf.float32, name="input"),)
    tf2onnx.convert.from_keras(model, input_signature=spec, opset=13, output_path=output_path)`

   I have then compiled openvino from source (github), using the command below:-
   [https://github.com/openvinotoolkit/openvino/wiki/BuildingForLinux](url)
   `cmake -DCMAKE_BUILD_TYPE=Release -DENABLE_INTEL_GPU=OFF -DENABLE_PYTHON=ON -DENABLE_WHEEL=ON ..`
   
Followed by installation of the wheel packages generated. and generation of IR using the below:-
    ` mo --input_model cv_experiments/model_1.onnx `

On running the IR's using the below code:-

    `import cv2
     opt_net = cv2.dnn.readNetFromModelOptimizer("model_1.xml","model_1.bin")`
     The following error is thrown:-

      "----> 1 opt_net = cv2.dnn.readNetFromModelOptimizer("model_1.xml","model_1.bin")

      error: Unknown model format! Cannot find reader for model format: xml and read the model: model_1.xml. Please check that reader library exists in your PATH."

   i have installed opencv using:-
   pip install opencv-python-inference-engine, It points to the latest opencv inference engine version.
   
   What could be the cause of this issue?

**Comment by jgespino** [comment]
@shubham162019 Since you built OpenVINO from source, most likely you built the latest master branch which is OpenVINO 2022.1. This version will generate IR v11 and may be incompatible with the version of OpenCV you are using. Please share the opencv version? The repo you linked is installing the following versions:
```
opencv-contrib-python==3.4.2.17
opencv-python==3.4.2.17
```

To confirm the models are working as expected, please try running your model using the benchmark_app
`benchmark_app -m model_1.xml`

Please also attach your onnx model and IR files(.bin & .xml) for me to test on my end.

Regards,
Jesus

**Comment by shubham162019** [comment]
Hi,
The opencv version the models are being tested on is :-  4.5.5 (obtained from `cv2.__version__`)
The output of `pip list` also confirms the same:- opencv-python-inference-engine 2022.1.5

The output logs for `mo --input_model ../../cv_experiments/model_1.onnx --input_shape [1,736,640,3] -o /datadrive/workspace/onnx2Ir` is as follows:-

Model Optimizer arguments:
Common parameters:
	- Path to the Input Model: 	/datadrive/workspace/openvino/openvino_output/../../cv_experiments/model_1.onnx
	- Path for generated IR: 	/datadrive/workspace/onnx2Ir
	- IR output name: 	model_1
	- Log level: 	ERROR
	- Batch: 	Not specified, inherited from the model
	- Input layers: 	Not specified, inherited from the model
	- Output layers: 	Not specified, inherited from the model
	- Input shapes: 	[1,736,640,3]
	- Source layout: 	Not specified
	- Target layout: 	Not specified
	- Layout: 	Not specified
	- Mean values: 	Not specified
	- Scale values: 	Not specified
	- Scale factor: 	Not specified
	- Precision of IR: 	FP32
	- Enable fusing: 	True
	- User transformations: 	Not specified
	- Reverse input channels: 	False
	- Enable IR generation for fixed input shape: 	False
	- Use the transformations config file: 	None
Advanced parameters:
	- Force the usage of legacy Frontend of Model Optimizer for model conversion into IR: 	False
	- Force the usage of new Frontend of Model Optimizer for model conversion into IR: 	False
OpenVINO runtime found in: 	/datadrive/workspace/openvino/openvino_output/python/python3.8/openvino
OpenVINO runtime version: 	custom_master_4da0941cd2e8f9829875e60df73d3cd01f820b9c
Model Optimizer version: 	custom_master_4da0941cd2e8f9829875e60df73d3cd01f820b9c
[ SUCCESS ] Generated IR version 11 model.
[ SUCCESS ] XML file: /datadrive/workspace/onnx2Ir/model_1.xml
[ SUCCESS ] BIN file: /datadrive/workspace/onnx2Ir/model_1.bin
[ SUCCESS ] Total execution time: 0.69 seconds. 
[ SUCCESS ] Memory consumed: 261 MB. 
[ INFO ] The model was converted to IR v11, the latest model format that corresponds to the source DL framework input/output format. While IR v11 is backwards compatible with OpenVINO Inference Engine API v1.0, please use API v2.0 (as of 2022.1) to take advantage of the latest improvements in IR v11.
Find more information about API v2.0 and IR v11 at https://docs.openvino.ai

The output logs for `benchmark_app -m /datadrive/workspace/onnx2Ir/model_1.xml` is as follows:-
[Step 1/11] Parsing and validating input arguments
[ WARNING ]  -nstreams default value is determined automatically for a device. Although the automatic selection usually provides a reasonable performance, but it still may be non-optimal for some cases, for more information look at README. 
[Step 2/11] Loading OpenVINO
[ WARNING ] PerformanceMode was not explicitly specified in command line. Device CPU performance hint will be set to THROUGHPUT.
[ INFO ] OpenVINO:
         API version............. custom_master_4da0941cd2e8f9829875e60df73d3cd01f820b9c
[ INFO ] Device info
         CPU
         openvino_intel_cpu_plugin version 2022.2
         Build................... custom_master_4da0941cd2e8f9829875e60df73d3cd01f820b9c

[Step 3/11] Setting device configuration
[ WARNING ] -nstreams default value is determined automatically for CPU device. Although the automatic selection usually provides a reasonable performance, but it still may be non-optimal for some cases, for more information look at README.
[Step 4/11] Reading network files
[ INFO ] Read model took 10.22 ms
[Step 5/11] Resizing network to match image sizes and given batch
[ INFO ] Network batch size: 1
[Step 6/11] Configuring input of the model
[ INFO ] Model input 'input' precision u8, dimensions ([N,H,W,C]): 1 736 640 3
[ INFO ] Model output 'conv2d_transpose_1' precision f32, dimensions ([...]): 1 736 640 1
[Step 7/11] Loading the model to the device
[ INFO ] Compile model took 148.28 ms
[Step 8/11] Querying optimal runtime parameters
[ INFO ] DEVICE: CPU
[ INFO ]   AVAILABLE_DEVICES  , ['']
[ INFO ]   RANGE_FOR_ASYNC_INFER_REQUESTS  , (1, 1, 1)
[ INFO ]   RANGE_FOR_STREAMS  , (1, 16)
[ INFO ]   FULL_DEVICE_NAME  , AMD EPYC 7R13 Processor
[ INFO ]   OPTIMIZATION_CAPABILITIES  , ['FP32', 'FP16', 'INT8', 'BIN', 'EXPORT_IMPORT']
[ INFO ]   CACHE_DIR  , 
[ INFO ]   NUM_STREAMS  , 4
[ INFO ]   AFFINITY  , Affinity.CORE
[ INFO ]   INFERENCE_NUM_THREADS  , 0
[ INFO ]   PERF_COUNT  , False
[ INFO ]   INFERENCE_PRECISION_HINT  , <Type: 'float32'>
[ INFO ]   PERFORMANCE_HINT  , PerformanceMode.THROUGHPUT
[ INFO ]   PERFORMANCE_HINT_NUM_REQUESTS  , 0
[Step 9/11] Creating infer requests and preparing input data
[ INFO ] Create 4 infer requests took 0.70 ms
[ WARNING ] No input files were given for input 'input'!. This input will be filled with random values!
[ INFO ] Fill input 'input' with random values 
[Step 10/11] Measuring performance (Start inference asynchronously, 4 inference requests using 4 streams for CPU, inference only: True, limits: 60000 ms duration)
[ INFO ] Benchmarking in inference only mode (inputs filling are not included in measurement loop).
[ INFO ] First inference took 263.13 ms
[Step 11/11] Dumping statistics report
Count:          484 iterations
Duration:       60723.14 ms
Latency:
    Median:     491.98 ms
    AVG:        501.47 ms
    MIN:        471.16 ms
    MAX:        756.12 ms
Throughput: 7.97 FPS

Please find the link below to the model repo which contains the following:-

https://primefocusmedia-my.sharepoint.com/:f:/g/personal/shubham_jaiswal_primefocus_com1/EiaXX-BHY6NMpd7ID6ianOgBciK_2QYHBjju1pbN4coavw?e=UNRAtB

1) model_1.onnx 
2) onnx2Ir (Folder containing the IR obtained from model_1.onnx)
3) working_Ir_for_opencv_4_3_0_2 (Folder containing working IRs for opencv-python-inference-engine==4.3.0.2, I had buit this in 2020, and donot have the .onnx/.pb files used to generate the IR)

**Comment by RakshaRamkumar** [comment]
@jgespino,

Even I am facing the same issue and error. My system also has the same version of OpenCV(4.5.5) and Openvino2022. The converted IR models are working (checked with benchmark_app) but the "Unknown model format!" error exists.

This error persists even for the models that I downloaded using omz_downloader.

I have also verified that the 'libinference_engine_ir_reader.so' and the 'libinference_engine.so' libraries are in the same folder.

Any suggestions/solutions to resolve this problem is appreciated.

Thanks,
Raksha

**Comment by jgespino** [comment]
@RakshaRamkumar @shubham162019 OpenVINO 2022.1 release will install OpenCV 4.5.5.64 that does not include the OpenCV DNN module. Since benchmark_app is able to run your models, I don't believe this is an OpenVINO issue.

If you want to use OpenCV DNN module it is recommended to build from source. Alternatively, you could post your question on the opencv-python-inference-engine repo since that is an unofficial pre-built of OpenCV.

https://github.com/banderlog/opencv-python-inference-engine

**Comment by RakshaRamkumar** [comment]
@jgespino ,

Thanks for the clarification.
